### PR TITLE
feat(es/minifier): Enable `reduce_vars` by default

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

This is to align with terser.

**Related issue:**

### CODE REVIEW

File: crates/swc_ecma_minifier/src/compress/optimize/strings.rs [hunk 1]
```diff
@@ -124,39 +124,6 @@ where
                 }));
             }
 
-            Expr::Ident(i) => {
```
Comment (kdy1) [inline crates/swc_ecma_minifier/src/compress/optimize/strings.rs:127]: This made output larger, so I removed this code

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc/tests/tsc-references/callWithSpreadES6.2.minified.js [hunk 2]
```diff
@@ -1,7 +1,6 @@
 //// [callWithSpreadES6.ts]
 var a, obj, xa;
-function foo(x, y, ...z) {}
-foo(1, 2, "abc"), foo(1, 2, ...a), foo(1, 2, ...a, "abc"), obj.foo(1, 2, "abc"), obj.foo(1, 2, ...a), obj.foo(1, 2, ...a, "abc"), obj.foo(1, 2, "abc"), obj.foo(1, 2, ...a), obj.foo(1, 2, ...a, "abc"), xa[1].foo(1, 2, "abc"), xa[1].foo(1, 2, ...a), xa[1].foo(1, 2, ...a, "abc"), xa[1].foo(...[
```
Comment (jridgewell) [inline crates/swc/tests/tsc-references/callWithSpreadES6.2.minified.js:4]: The `...a` spread here is observable, and the `...z` rest is also observable.

File: crates/swc/tests/tsc-references/taggedTemplateStringsWithOverloadResolution1_ES6.2.minified.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
 //// [taggedTemplateStringsWithOverloadResolution1_ES6.ts]
 function foo(...stuff) {}
-foo([]), foo([], 1), foo([], 1, 2), foo([], 1, !0), foo([], 1, "2"), foo([], 1, 2, 3), foo``, foo`${1}`, foo`${1}${2}`, foo`${1}${!0}`, foo`${1}${"2"}`, foo`${1}${2}${3}`;
+foo``, foo`${1}`, foo`${1}${2}`, foo`${1}${!0}`, foo`${1}${"2"}`, foo`${1}${2}${3}`;
```
Comment (jridgewell) [inline crates/swc/tests/tsc-references/taggedTemplateStringsWithOverloadResolution1_ES6.2.minified.js:3]: Want to optimize this, too?

**Review by jridgewell — APPROVED** [review]

**Pull request merged**